### ISSUE TITLE
Proposal: use __repr__ as well as __str__? Or IPython display?

### ISSUE THREAD

**Issue by jabooth** [menpo/menpo#752]
I can't think of the last time I made use of the current repr output, e.g. for an image:
```
<menpo.image.base.Image at 0x7f5d8f3c3fd0>
```
I never look at the object adress here. It's only use is really diagnosing if two variables refer to the same object, but I would much prefer `a is b` to test that rather than looking for differences in these ids.

The class name is useful, but we also typically report this in our `__str__` representations.

It would be much more practical to just get our current `__str__` rep in the notebook when you evaluate an object with no print statement (as numpy does with it's arrays for example). Maybe we should just have `__str__` and `__repr__` return the same thing for our objects?

I note that numpy for instance, _doesn't_ return the same, instead providing a cleaner output for `__str__` vs `__repr__`:

```
__repr__(): array([ 1.,  1.])
```
```
__str__(): [ 1.  1.]
```
But I'm not sure if we really care to make such a distinction.

I guess the alternative or complementary proposal to this would to be more aggressive in showing a user an IPython widget or matplotlib inline representation of an object when they evaluate an object alone. This I think might be a step too far though and hard to define sensible behavior for (I don't want a full blown mayavi visualization of a mesh firing up in a new window every time I evaluate a mesh for instance).

**Comment by patricksnape** [comment]
Where is the documentation on the IPython display magic stuff?

**Comment by jabooth** [comment]
http://ipython.readthedocs.io/en/stable/config/integrating.html?highlight=display#rich-display

**Comment by jabooth** [comment]
hmm the custom tab completion could be interesting too..?

**Comment by patricksnape** [comment]
Yeah I can get behind this

**Comment by jabooth** [comment]
Closed by #782
